include/dicom/dicom.h

```diff
@@ -220,1618 +272,586 @@ typedef struct _DcmError DcmError;
+ * :param vm: Number of values
+ * :param steal: if true, ownership of values passes to element
  *
- * :return: Pointer to memory location where value is stored
+ * :return: true on success
  */
 DCM_EXTERN
-const char *dcm_element_get_value_OB(const DcmElement *element);
+bool dcm_element_set_value_string_multi(DcmError **error,
+                                        DcmElement *element,
+                                        char **values,
+                                        uint32_t vm,
+                                        bool steal);
 
 /**
- * Get value of a Data Element with Value Representation OD (Other Double).
+ * Get an integer from a 16, 32 or 64-bit integer-valued element.
  *
+ * :param error: Error structure pointer
  * :param element: Pointer to Data Element
+ * :param index: Zero-based index of value within the Data Element
+ * :param value: Integer value return
  *
- * :return: Pointer to memory location where value is stored
+ * :return: true on success
  */
 DCM_EXTERN
-const char *dcm_element_get_value_OD(const DcmElement *element);
+bool dcm_element_get_value_integer(DcmError **error,
+                                   const DcmElement *element,
+                                   uint32_t index,
+                                   int64_t *value);
 
 /**
- * Get value of a Data Element with Value Representation OF (Other Float).
+ * Set the value of an element to an integer. The element must have an
+ * appropriate tag.
  *
+ * :param error: Error structure pointer
  * :param element: Pointer to Data Element
+ * :param value: Integer value
  *
- * :return: Pointer to memory location where value is stored
+ * :return: true on success
  */
 DCM_EXTERN
-const char *dcm_element_get_value_OF(const DcmElement *element);
+bool dcm_element_set_value_integer(DcmError **error,
+                                   DcmElement *element,
+                                   int64_t value);
 
 /**
- * Get value of a Data Element with Value Representation OL (Other Long).
+ * Set the value of an element to an array of numeric values. The element must
+ * have an appropriate tag.
+ *
+ * Although the value passed is `int*`, it should
+ * be a pointer to an array of 16- to 64-bit numeric values of the
+ * appropriate type for the element VR.
  *
+ * On success, if `steal` is true, ownership of `values` passes to
+ * `element`, ie. it will be freed when `element` is destroyed. If `steal` is
+ * false, then a copy is made of `values` and ownership is not transferred.
+ *
+ * :param error: Error structure pointer
  * :param element: Pointer to Data Element
+ * :param values: Array of values
+ * :param vm: Number of values
+ * :param steal: if true, ownership of values passes to element
  *
- * :return: Pointer to memory location where value is stored
+ * :return: true on success
  */
 DCM_EXTERN
-const char *dcm_element_get_value_OL(const DcmElement *element);
+bool dcm_element_set_value_numeric_multi(DcmError **error,
```

Comment:
It'll do any numeric VR, it just takes an int pointer.

Reply:
Would it make sense to have different functions for integer and floating-point numbers? We have separate setter functions for the single value case as well as separate getter functions.

I like the idea of abstracting DICOM VRs. However, I think the distinction between integer and floating-point numeric values at the API level would be meaningful.

Reply:
I think it would just add extra API and code paths that would need testing, there would be no benefit. 

How about swapping that `int*` pointer for a `void*`? That would make it clear that it needed to be cast to the appropriate numeric type before access.

Reply:
I changed it to `void*`.

Reply:
I am generally not a fan of void pointers in the public API? Is it necessary to expose this function publicly or could it only be used under the hood? Something to consider for future versions?

Reply:
It should be very handy for eg. a python binding --- to implement something like:

```python
element.set([1, 2, 3])
```

We would only need this single API point for all numeric array types.